client/src/main.rs

```diff
@@ -1405,20 +1418,35 @@ struct Record {
 /**
  * A simple IO test in two stages: 100% random writes, then 100% random
  * reads. The caller can select:
- * io_depth: The number of outstanding IOs issued at a time
+ * io_depth:      The number of outstanding IOs issued at a time
  * blocks_per_io: The size of each io (in multiple of block size).
- * count: The number of loops to perform for each test (all IOs in io_depth
- *        are considered as a single loop).
+ * count:         The number of loops to perform for each test (all IOs
+ *                in io_depth are considered as a single loop).
+ * write_loop:    The number of times to do the 100% write loop.
+ * read_loop:     The number of times to do the 100% read loop.
+ *
  * A summary is printed at the end of each stage.
  */
+#[allow(clippy::too_many_arguments)]
 async fn perf_workload(
     guest: &Arc<Guest>,
     ri: &mut RegionInfo,
     wtr: &mut Option<&mut csv::Writer<File>>,
     count: usize,
     io_depth: usize,
     blocks_per_io: usize,
+    write_loop: usize,
+    read_loop: usize,
 ) -> Result<()> {
+    // Before we start, make sure the work queues are empty.
+    loop {
+        let wc = guest.query_work_queue()?;
+        if wc.up_count + wc.ds_count == 0 {
+            break;
+        }
+        std::thread::sleep(std::time::Duration::from_secs(2));
```

Comment:
I think these should be tokio::time::sleep(...).await instead? this blocks the whole thread

Reply:
Updated.